src/test/unit/EigenPod-PodManagerUnit.t.sol

```diff
@@ -1,53 +1,666 @@
+        eigenPod.verifyAndProcessWithdrawals(
+            0,
+            stateRootProofStruct,
+            withdrawalProofsArray,
+            validatorFieldsProofArray,
+            validatorFieldsArray,
+            withdrawalFieldsArray
+        );
+    }
+
+    function test_verifyWithdrawalCredentials_revert_pausedEigenVerifyCredentials() public {
+        BeaconChainProofs.StateRootProof memory stateRootProofStruct;
+
+        bytes32[][] memory validatorFieldsArray = new bytes32[][](1);
+        bytes[] memory proofsArray = new bytes[](1);
+        uint40[] memory validatorIndices = new uint40[](1);
+
+        // pause the contract
+        cheats.prank(address(pauser));
+        eigenPodManager.pause(2 ** PAUSED_EIGENPODS_VERIFY_CREDENTIALS);
+
+        cheats.prank(address(podOwner));
+        cheats.expectRevert(bytes("EigenPod.onlyWhenNotPaused: index is paused in EigenPodManager"));
+        eigenPod.verifyWithdrawalCredentials(
+            0,
+            stateRootProofStruct,
+            validatorIndices,
+            proofsArray,
+            validatorFieldsArray
+        );
+    }
+
+    function test_activateRestaking_revert_pausedEigenVerifyCredentials() public {
+        // pause the contract
+        cheats.prank(address(pauser));
+        eigenPodManager.pause(2 ** PAUSED_EIGENPODS_VERIFY_CREDENTIALS);
+
+        cheats.prank(address(podOwner));
+        cheats.expectRevert(bytes("EigenPod.onlyWhenNotPaused: index is paused in EigenPodManager"));
+        eigenPod.activateRestaking();
+    }
+}
+
+contract EigenPod_PodManager_UnitTests_EigenPod is EigenPod_PodManager_UnitTests {
+    /**
+     * @notice Tests function calls from EPM to EigenPod
+     * 1. Stake works when pod is deployed
+     * 2. Stake when pod is not deployed -> check that ethPOS deposit contract is correct for this and above test
+     * 3. Withdraw restakedBeaconChainETH -> ensure we do a full withdrawal before
+     */
+
+    function test_stake_podAlreadyDeployed() public {
+        bytes memory pubkey = hex"88347ed1c492eedc97fc8c506a35d44d81f27a0c7a1c661b35913cfd15256c0cccbd34a83341f505c7de2983292f2cab";
+    
+        bytes memory signature;
+    
+        bytes32 depositDataRoot;
+    
+        uint256 stakeAmount = 32e18;
+
+        uint256 maxPods = eigenPodManager.maxPods();
+        uint256 numPods = eigenPodManager.numPods();
+        emit log_named_uint("maxPods", maxPods);
+        emit log_named_uint("numPods", numPods);
+
+        cheats.startPrank(podOwner);
+        cheats.deal(podOwner, stakeAmount);
+        eigenPodManager.stake{value: stakeAmount}(pubkey, signature, depositDataRoot);
+        cheats.stopPrank();
+    }
+
+    function test_stake_podNotDeployed() public {
+        address newPodOwner = address(69696969696);
+        bytes memory pubkey = hex"88347ed1c492eedc97fc8c506a35d44d81f27a0c7a1c661b35913cfd15256c0cccbd34a83341f505c7de2983292f2cab";
```

Comment:
Nit: pub key, signature, depositdataRoot can be state variables declares in this test subcontract so you don't have to redeclare in each test here